package/src/components/ChannelList/ChannelListMessenger.tsx

```diff
@@ -151,14 +152,12 @@ const ChannelListMessengerWithContext = <
   }
 
   const onEndReached = () => {
-    if (loadNextPage) {
+    if (!onEndReachedCalledDuringMomentumRef.current && hasNextPage) {
       loadNextPage();
+      onEndReachedCalledDuringMomentumRef.current = true;
```

Comment:
A suggestion, can we rename `onEndReachedCalledDuringMomentumRef` to `onEndReachedCalledDuringCurrentScrollRef`... better understandable I guess